components/playback/controls/player/ButtonWidget.vue

```diff
@@ -0,0 +1,199 @@
+    <span
+      v-for="btn_index in num_of_btn"
+      :key="btn_index"
+      :ref="btn_index.toString()"
+      class="span__button_label"
+      :style="btn_stateprop(btn_index)"
+      @click="selected(btn_index)"
+      @mouseenter="hover_active(btn_index)"
+      @mouseleave="hover_inactive(btn_index)"
+      >{{ button_names[btn_index - 1] }}
+    </span>
+    <div v-if="num_of_verticalline >= 1">
+      <canvas
+        v-for="line_index in num_of_verticalline"
+        :key="line_index"
+        class="canvas__vertical-line"
+        :style="btn_divider_display(line_index)"
+      >
+      </canvas>
+    </div>
+    <canvas
+      class="canvas__common-horizontal-line"
+      :style="'width: ' + (button_widget_width - 10) + 'px;'"
+    >
+    </canvas>
+  </div>
+</template>
+<script>
+export default {
+  name: "ButtonWidget",
+  props: {
+    button_names: {
+      type: Array,
+      required: true,
+    },
+    enabled_color: { type: String, default: "" },
+    disabled_color: { type: String, default: "" },
+    hover_color: {
+      type: Array,
+      required: true,
+    },
+    is_enabled: {
+      type: Array,
+      default: function () {
+        return new Array(this.num_of_btn).fill(true);
+      },
+    },
+    button_widget_width: { type: Number, default: 0 },
+    button_widget_height: { type: Number, default: 0 },
+    button_widget_top: { type: Number, default: 0 },
+    button_widget_left: { type: Number, default: 0 },
+  },
+  data() {
+    return {
+      count: 0,
+    };
+  },
+  computed: {
+    num_of_btn: function () {
+      return this.button_names.length;
+    },
+    num_of_verticalline: function () {
+      return this.button_names.length - 1;
+    },
+    background_cssprops: function () {
+      return (
+        "width: " +
+        this.button_widget_width +
+        "px; height: " +
+        this.button_widget_height +
+        "px; top: " +
+        this.button_widget_top +
+        "px; left: " +
+        this.button_widget_left +
+        "px;"
+      );
+    },
+  },
+  methods: {
+    btn_stateprop(value) {
+      this.count = value - 1;
+      const computed_width = this.button_widget_width / this.num_of_btn;
+      const computed_left = computed_width * this.count;
+      return this.is_enabled[this.count]
+        ? "color: " +
+            this.enabled_color +
+            ";" +
+            "width: " +
+            computed_width +
+            "px;" +
+            "left: " +
+            computed_left +
+            "px;"
+        : "color: " +
+            this.disabled_color +
+            ";" +
+            "width: " +
+            computed_width +
+            "px;" +
+            "left: " +
+            computed_left +
+            "px;";
+    },
+    btn_divider_display(value) {
+      const computed_width = this.button_widget_width / this.num_of_btn;
+      if (value == 0) {
+        return "left: " + computed_width + "px;";
```

Comment:
this line never gets executed, because the way you have your `v-for` loop set up, `value` will never be `0`.  Are you sure this is doing the calculations the way you intended?

Reply:
Yes, this conditional check is not needed, earlier during the TDD based development, it was possible to have v-if and v-for to be included in the <canvas>, so this line was needed as v-for entry was happening once.

Eslint executing threw an error "vue/no-use-v-if-with-v-for" so this resulted to move v-if at div level, on the function this line is not relevant, removed. 

On the logic part the following are sequences
a) Single Button then (no vertical line) as 'v-if=num_of_verticalline >= 1'  would be false so no need to have vertical line.
b) More than One Button (vertical line) as 'v-if=num_of_verticalline > -1' would be true so a based on the number of buttons vertical line with right breakdown of offset either 1/2 for two buttons and  1/3 or 2/3 for three buttons.